interfaces/kalosm/src/surrealdb_integration/hybrid_search.rs

```diff
@@ -0,0 +1,1072 @@
+    /// * `results` - Maximum number of results to return
+    pub fn with_results(mut self, results: usize) -> Self {
+        self.results = results;
+        self
+    }
+
+    /// Set the weight of the semantic search
+    ///
+    /// The keyword weight will be automatically adjusted to `1.0 - weight`
+    /// to ensure the weights sum to 1.0.
+    ///
+    /// Default: 0.7
+    ///
+    /// # Arguments
+    /// * `weight` - Weight for semantic score (0.0 to 1.0)
+    ///
+    /// # Example
+    /// ```
+    /// // Setting semantic weight to 0.8 will set keyword weight to 0.2
+    /// builder.with_semantic_weight(0.8)
+    /// ```
+    pub fn with_semantic_weight(mut self, weight: f32) -> Self {
+        self.semantic_weight = weight.clamp(0.0, 1.0);
+        self.keyword_weight = 1.0 - self.semantic_weight;
+        self
+    }
+
+    /// Set the weight of the keyword search
+    ///
+    /// The semantic weight will be automatically adjusted to `1.0 - weight`
+    /// to ensure the weights sum to 1.0.
+    ///
+    /// Default: 0.3
+    ///
+    /// # Arguments
+    /// * `weight` - Weight for keyword score (0.0 to 1.0)
+    ///
+    /// # Example
+    /// ```
+    /// // Setting keyword weight to 0.4 will set semantic weight to 0.6
+    /// builder.with_keyword_weight(0.4)
+    /// ```
+    pub fn with_keyword_weight(mut self, weight: f32) -> Self {
+        self.keyword_weight = weight.clamp(0.0, 1.0);
+        self.semantic_weight = 1.0 - self.keyword_weight;
+        self
+    }
+
+    /// Set the RRF constant k
+    ///
+    /// Lower values give more weight to higher-ranked results.
+    /// Typical values range from 1 to 100, with 60 being standard.
+    ///
+    /// Default: 60.0
+    ///
+    /// # Arguments
+    /// * `k` - RRF constant k
+    pub fn with_rrf_k(mut self, k: f32) -> Self {
+        self.rrf_k = k.clamp(1.0, 100.0);
+        self
+    }
+
+    /// Execute the hybrid search using weighted score combination
+    ///
+    /// This method normalizes both semantic and keyword scores to 0-1 range,
+    /// then combines them using the configured weights.
+    ///
+    /// # Returns
+    /// Vector of search results sorted by combined score
+    ///
+    /// # Errors
+    /// Returns `HybridSearchError` if the search fails
+    pub async fn run_weighted(self) -> Result<Vec<HybridSearchResult<R>>, HybridSearchError>
+    where
+        R: Serialize + DeserializeOwned + Clone + AsRef<Document> + Send + Sync,
+        <M as Embedder>::Error: std::fmt::Debug + std::fmt::Display + 'static,
+    {
+        let search_limit = self.results * 3;
+
+        let semantic_results = self
+            .table
+            .search(self.user_query.clone())
+            .with_results(search_limit)
+            .run()
+            .await?;
+
+        let links_table = self.table.table().table_links();
+        let keyword_query = r#"
+            SELECT
+                document_id,
+                search::score(1) as keyword_score
```

Comment:
This always uses the relevance score for the second field right? It should depend on the (currently unused field_name)

Reply:
Hey hey,
These queries are using the `@@` operator, which matches the same number passed into the `search::score()` function. In these cases, it's the `chunk_text` field from the linking table.

The `field_name` is used while creating the `HybridDocumentTable` builder, which creates the setup for the search index.

Reply:
Thanks, I missed the usage of field_name when building the extraction function